pkg/yurtmanager/webhook/deploymentrender/v1alpha1/deploymentrender_default.go

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright 2023 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the License);
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an AS IS BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/apps/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openyurtio/openyurt/pkg/apis/apps/v1alpha1"
+	"github.com/openyurtio/openyurt/pkg/yurtmanager/controller/yurtappset/adapter"
+)
+
+const (
+	DeploymentMutatingWebhook = "deployment-webhook"
+)
+
+var (
+	resources = []string{"YurtAppSet", "YurtAppDaemon"}
+)
+
+func contain(kind string, resources []string) bool {
+	for _, v := range resources {
+		if kind == v {
+			return true
+		}
+	}
+	return false
+}
+
+// Default satisfies the defaulting webhook interface.
+func (webhook *DeploymentRenderHandler) Default(ctx context.Context, obj runtime.Object) error {
+	deployment, ok := obj.(*v1.Deployment)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a YurtAppOverrider but got a %T", obj))
+	}
+	if deployment.OwnerReferences == nil {
+		return nil
+	}
+	if !contain(deployment.OwnerReferences[0].Kind, resources) {
+		return nil
+	}
+	klog.Info("start to validate deployment")
+	// Get YurtAppSet/YurtAppDaemon resource of this deployment
+	app := deployment.OwnerReferences[0]
+	var instance client.Object
+	switch app.Kind {
+	case "YurtAppSet":
+		instance = &v1alpha1.YurtAppSet{}
+	case "YurtAppDaemon":
+		instance = &v1alpha1.YurtAppDaemon{}
+	default:
+		return nil
+	}
+	if err := webhook.Client.Get(ctx, client.ObjectKey{
+		Namespace: deployment.Namespace,
+		Name:      app.Name,
+	}, instance); err != nil {
+		return err
+	}
+	klog.Info("Successfully get the owner of deployment")
```

Comment:
please delete this dummy log: klog.Info("Successfully get the owner of deployment")